libglusterfs/src/inode.c

```diff
@@ -580,7 +577,7 @@ __inode_ref(inode_t *inode, bool is_invalidate)
      * in inode table increases which is wrong. So just keep the ref
      * count as 1 always
      */
-    if (__is_root_gfid(inode->gfid) && inode->ref)
+    if (inode->ref && __is_root_gfid(inode->gfid))
```

Comment:
nit (in a different patch!) we can do the is root gfid outside this code path, to the best of my knowledge.

Reply:
Yes. That's what I said. However now I've realized that `__inode_ref()` and `_inode_unref()` are declared `static`, which I missed in the other review, so their scope is very limited and we could move the check to the callers from this same module. In this case it's under control. I'll update the other PR.